pkg/controllerhelpers/handlers.go

```diff
@@ -90,14 +90,20 @@ func (h *Handlers[T]) Enqueue(depth int, untypedObj kubeinterfaces.ObjectInterfa
 func (h *Handlers[T]) EnqueueAll(depth int, untypedObj kubeinterfaces.ObjectInterface, op HandlerOperationType) {
 	klog.V(4).InfoSDepth(depth, "Enqueuing all controller objects", getObjectLogContext(untypedObj, nil)...)
 
-	controllerObjs, err := h.getterLister.List(untypedObj.GetNamespace(), labels.Everything())
-	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("couldn't list all controller objects for %T: %w", untypedObj, err))
-		return
-	}
+	h.EnqueueAllFunc(h.Enqueue)(depth+1, untypedObj, op)
+}
+
+func (h *Handlers[T]) EnqueueAllFunc(enqueueFunc EnqueueFuncType) EnqueueFuncType {
```

Comment:
I think we always want to call the `.Enqueue` function. The goal from  https://github.com/scylladb/scylla-operator-enterprise/pull/25#discussion_r1638032987 seems to be more like `EnqueueIf`

/lgtm cancel

Reply:
> I think we always want to call the .Enqueue function. 

Why?

This is aligned with existing `EnqueueOwner` and `EnqueueOwnerFunc` and imo allows for greater flexibility than introducing something like `EnqueueAllWithFilter` (I guess this is what you had in mind by EnqueueIf?), since you can achieve that with function composition then.

Reply:
I haven't actually expanded all the code around, you are right that this is in line with the other patterns there.
/lgtm